src/sidebar/util/type-coercions.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Type conversion methods that gently coerce values that other parts
+ * of the UI may make assumptions on. This is needed for incoming parameters
+ * that are otherwise not sanitized. Note that these methods do not throw 
+ * errors or act like validators with mal-formed values.
```

Comment:
The "do not throw errors" part assumes that the values are "plain JavaScript values", ie. values that could be produced by `JSON.parse`. This is true in the case of configuration parsed in `host-config.js`. I think it would be worth noting that assumption though.